models/keyword.go

```diff
@@ -18,28 +18,62 @@ type Keyword struct {
 
 	Keyword string `orm:"size(128)"`
 	Url     string `orm:"size(128)"`
+	Status  int    `orm:"default(0)"`
 }
 
+var statusKeyword = map[string]int{"failed": -1, "pending": 0, "completed": 1}
+
 func init() {
 	orm.RegisterModel(new(Keyword))
 }
 
-// GetAllKeyword retrieves all Keyword matches certain condition and returns empty list if no records exist.
+// AddKeyword inserts a new Keyword into database and returns last inserted Id on success.
+func AddKeyword(keyword *Keyword) (id int64, err error) {
+	ormer := orm.NewOrm()
+	id, err = ormer.Insert(keyword)
+
+	return id, err
+}
+
+// GetKeyword retrieves a Keyword by matching with certain conditions and returning error if it doesn't exist
+func GetKeyword(query map[string]interface{}, orderByList []string) (keyword *Keyword, err error) {
```

Comment:
Just minor naming preference for `GetKeywordBy` as for `GetKeyword` I would expect to have an ID as parameter.